Modules/xASL_module_BIDS2Legacy.m

```diff
@@ -83,18 +80,20 @@
         % x.modules.bids2legacy.BIDS.subjects(1).perf(1) -> filename: 'sub-10015124_ses-1_run-1_asl.nii.gz'
         % x.modules.bids2legacy.BIDS.subjects(1).perf(2) -> filename: 'sub-10015124_ses-1_run-2_asl.nii.gz'
         % x.modules.bids2legacy.BIDS.subjects(1).perf(3) -> filename: 'sub-10015124_ses-1_run-1_m0scan.nii.gz'
-        % x.modules.bids2legacy.BIDS.subjects(1).perf(4) -> filename: 'sub-10015124_ses-1_run-2_m0scan.nii.gz'            
+        % x.modules.bids2legacy.BIDS.subjects(1).perf(4) -> filename: 'sub-10015124_ses-1_run-2_m0scan.nii.gz'  
+		%
+		% Finally note that visits can now be any name, number, data, or a string and are always treated as a string
 
         % Subject ID
         SubjectID = x.modules.bids2legacy.BIDS.subjects(iSubjSess).name;
         SessionID = x.modules.bids2legacy.BIDS.subjects(iSubjSess).session;
         if isempty(SessionID)
-            iVisit = 1; % we default to a single visit
+            SessionID = '1';
         else
-            iVisit = str2num(SessionID(5:end));
+            SessionID = xASL_adm_CorrectName(SessionID(5:end), 2);
```

Comment:
Have to issue a warning if something is corrected

Reply:
I agree. Note that we are correcting the names while loading the list of session and while doing the conversion. We should only report the warning on a single location, so I would do it here, during the actual renaming, rather than when creating the list of subjects. Do you agree?

Reply:
Sounds good. Didn't xASL_adm_CorrectName already have an input boolean for issuing a warning or not when something is changed? Otherwise we could add this, because this is of course something that happens a lot; there are probably many cases where you want the user to know that a name is changed.

Reply:
Done

Reply:
???

I blindly accepted this because you said done, but I don't see where you have done this?

It seems you have now only done the first thing mentioned here (adding the xASL_adm_CorrectName), but didn't read my full message???

I provided a similar message in another function in this review about adding a warning boolean to xASL_adm_CorrectName, we may want to activate this here as well.

Are you in a rush when reading my comments? Doesn't really help except frustration ;)